### PR TITLE
Update react-testing.md

### PR DESCRIPTION
**Pull request by JeffreyATW**

### CODE REVIEW

File: languages/javascript/react-testing.md [hunk 1]
```diff
@@ -2,95 +2,16 @@
...
+## Unit and integration testing: Jest
 
-## Unit testing: Mocha
+We use [Jest](https://jestjs.io/) as our testing framework. We make use of Jest's assertions, matchers, mocks, built-in code coverage, and more.
 
-We use [Mocha](https://mochajs.org/) as our unit-testing framework. It provides us with an easy command-line interface, hooks, and dead-simple callbacks for asynchronous tests.
+We use Jest to write unit tests for code outside of React components, such as API routes and helpers. We also use Jest to write integration tests with the help of React Testing Library.
 
-In order to test transpiled files, we need to pass [`babel-register`](https://babeljs.io/docs/usage/require/) as a compiler, like so:
+We often strive toward 100% code coverage when reasonable. If necessary due to JSDOM limitations, third-party code which is hard to partially mock, or Jest's inability to pass tests with uncaught errors, we tell Istanbul, Jest's code coverage tool, to `istanbul ignore` particular lines, followed by a descriptive comment explaining why.
```
Comment (tgaff) [inline languages/javascript/react-testing.md:13]: This sentence here is pretty long and hard to parse.  Might be better split up into multiple sentences. 
I also like how the previous version discussed code-coverage in its own section.

File: languages/javascript/react-testing.md [hunk 2]
```diff
@@ -2,95 +2,16 @@
 
 ## Overview
 
-The React testing ecosystem is constantly evolving, and like many areas of the Node world, it is made up of many small tools rather than an all-encompassing test-running solution. This guide is intended to describe what decisions we've made for unit-testing our React apps, and is not intended to be a thorough how-to guide on setting up an app for testing. It should be seen solely as a snapshot of what we're doing as of mid 2016.
+We're old hands at React testing, having lived through the eras of Jasmine, Mocha, Chai, Sinon, Enzyme, et cetera. But testing has evolved mostly thanks to Jest and React Testing Library covering all of these cases of these older, smaller tools, and so have we!
```
[the remark this answers is not in the document]
Comment (tgaff) [reply languages/javascript/react-testing.md:5]: The last sentence has at least a grammar issue and is a little bit run-on-ish.  Maybe we can break it up?

```suggestion
We're old hands at React testing, having lived through the eras of Jasmine, Mocha, Chai, Sinon, Enzyme, et cetera. Over time testing practices have evolved significantly, primarily with the emergence of Jest and React Testing Library, which now encompass the capabilities of these older, smaller tools. We too have evolved and embrace RTL and Jest as our standard test setup.
```

**Review by tgaff — DISMISSED** [review]
1 additional suggestion, but otherwise lgtm

**Review by holetse — APPROVED** [review]
Thanks Jeffrey!

**Review by tgaff — APPROVED** [review]
